packages/clerk-js/src/core/resources/User.ts

```diff
@@ -177,6 +180,17 @@ export class User extends BaseResource implements UserResource {
     return new DeletedObject(json);
   };
 
+  createBackupCode = async (): Promise<BackupCodeResource> => {
+    const json = (
+      await BaseResource._fetch<BackupCodeJSON>({
+        path: this.path() + '/backup_codes/',
+        method: 'POST',
+      })
+    )?.response as unknown as BackupCodeJSON;
```

Comment:
This is not an issue with this specific PR, but the type casting/ checking for the `verifyTOTP`, `disableTOTP` and `createBackupCode` is definitely brittle and since it involves network requests, the actual runtime value might be different that what we expect at this point

I did a very quick pass and I think that instead of the manual type casts,  I'd rather use an assertion function to at least throw/ try to recover at that point. What do you think?

This can definitely be done in a separate PR as its not related with the changes made by Haris :)

cc @SokratisVidros

Reply:
+1